components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/JSON.kt

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.ktx.kotlin
+
+import org.json.JSONArray
+import org.json.JSONObject
+import java.util.TreeMap
+
+@Suppress("UNCHECKED_CAST")
+fun <T> JSONArray?.toList(): List<T> {
+    if (this != null) {
+        val result = ArrayList<T>()
+        for (i in 0 until length())
+            result.add(get(i) as T)
+        return result
+    }
+    return listOf()
+}
+
+fun JSONObject.tryGetString(key: String): String? {
```

Comment:
Let's try to add some JavaDoc .. so that a developer understands the difference between `getString`, `optString` and `tryGetString`.